src/entities/FormattingNode/index.spec.ts

```diff
@@ -0,0 +1,281 @@
+
+describe('FormattingNode', () => {
+  const childMock = createChildMock('Some text here. ');
+  const anotherChildMock = createChildMock('Another text here.');
+
+  const tool = createInlineToolName('bold');
+  const anotherTool = createInlineToolName('italic');
+  const data = createInlineToolData({});
+  let node: FormattingNode;
+
+  beforeEach(() => {
+    node = new FormattingNode({
+      tool,
+      data,
+      parent: parentMock as FormattingNode,
+      children: [childMock, anotherChildMock],
+    });
+
+    jest.clearAllMocks();
+  });
+
+  describe('.length', () => {
+    it('should return length sum of lengths of children', () => {
+      expect(node.length).toEqual(childMock.length + anotherChildMock.length);
+    });
+  });
+
+  describe('.serialized()', () => {
+    it('should serialized node\'s values', () => {
+      const result = node.serialized;
+
+      expect(result).toEqual({
+        text: childMock.getText() + anotherChildMock.getText(),
+        fragments: [
+          {
+            tool,
+            data,
+            range: [0, node.length],
+          },
+        ],
+      });
+    });
+  });
+
+  describe('.insertText()', () => {
+    const newText = 'new text';
+    const index = 3;
+
+    it('should call insertText() of child by the passed index', () => {
+      node.insertText(newText, index);
+
+      expect(childMock.insertText).toBeCalledWith(newText, index);
+    });
+
+    it('should adjust index by child offset', () => {
+      const offset = childMock.length;
+
+      node.insertText(newText, index + offset);
+
+      expect(anotherChildMock.insertText).toBeCalledWith(newText, index);
+    });
+
+    it('should append text to the last child by default', () => {
+      node.insertText(newText);
+
+      expect(anotherChildMock.insertText).toBeCalledWith(newText, anotherChildMock.length);
+    });
+  });
+
+  describe('.removeText()', () => {
+    const start = 3;
+    const end = 5;
+
+    it('should remove text from appropriate child', () => {
+      node.removeText(start, end);
+
+      expect(childMock.removeText).toBeCalledWith(start, end);
+    });
+
+    it('should adjust indices by child offset', () => {
+      const offset = childMock.length;
+
+      node.removeText(offset + start, offset + end);
+
+      expect(anotherChildMock.removeText).toBeCalledWith(start, end);
+    });
+
+    it('should call removeText for each affected child', () => {
+      const offset = childMock.length;
+
+      node.removeText(start, offset + end);
+
+      expect(childMock.removeText).toBeCalledWith(start, offset);
+      expect(anotherChildMock.removeText).toBeCalledWith(0, end);
+    });
+
+    it('should remove all text by default', () => {
+      node.removeText();
+
+      expect(childMock.removeText).toBeCalledWith(0, childMock.length);
+      expect(anotherChildMock.removeText).toBeCalledWith(0, anotherChildMock.length);
+    });
+
+    it('should call remove() if length is 0', () => {
```

Comment:
length of original text or text after the `removeText` call?

Reply:
After removeText call